collector/src/lib.rs

```diff
@@ -135,99 +134,32 @@ static COLLECTOR_USER_AGENT: &str = concat!(
     "collector"
 );
 
-/// The DAP collector's view of task parameters.
-#[derive(Derivative)]
-#[derivative(Debug)]
-pub struct CollectorParameters {
-    /// Unique identifier for the task.
-    task_id: TaskId,
-    /// The base URL of the leader's aggregator API endpoints.
-    #[derivative(Debug(format_with = "std::fmt::Display::fmt"))]
-    leader_endpoint: Url,
-    /// The authentication information needed to communicate with the leader aggregator.
-    authentication: AuthenticationToken,
-    /// HPKE keypair used for decryption of aggregate shares.
-    #[derivative(Debug = "ignore")]
-    hpke_keypair: HpkeKeypair,
-    /// Parameters to use when retrying HTTP requests.
-    #[derivative(Debug = "ignore")]
-    http_request_retry_parameters: ExponentialBackoff,
-    /// Parameters to use when waiting for a collection job to be processed.
-    #[derivative(Debug = "ignore")]
-    collect_poll_wait_parameters: ExponentialBackoff,
-}
-
-impl CollectorParameters {
-    /// Creates a new set of collector task parameters.
-    pub fn new(
-        task_id: TaskId,
-        leader_endpoint: Url,
-        authentication: AuthenticationToken,
-        hpke_keypair: HpkeKeypair,
-    ) -> CollectorParameters {
-        CollectorParameters {
-            task_id,
-            leader_endpoint: url_ensure_trailing_slash(leader_endpoint),
-            authentication,
-            hpke_keypair,
-            http_request_retry_parameters: http_request_exponential_backoff(),
-            collect_poll_wait_parameters: ExponentialBackoff {
-                initial_interval: StdDuration::from_secs(15),
-                max_interval: StdDuration::from_secs(300),
-                multiplier: 1.2,
-                max_elapsed_time: None,
-                ..Default::default()
-            },
-        }
-    }
-
-    /// Replace the exponential backoff settings used for HTTP requests.
-    pub fn with_http_request_backoff(mut self, backoff: ExponentialBackoff) -> CollectorParameters {
-        self.http_request_retry_parameters = backoff;
-        self
-    }
-
-    /// Replace the exponential backoff settings used while polling for aggregate shares.
-    pub fn with_collect_poll_backoff(mut self, backoff: ExponentialBackoff) -> CollectorParameters {
-        self.collect_poll_wait_parameters = backoff;
-        self
-    }
-
-    /// Construct a URI for a collection.
-    fn collection_job_uri(&self, collection_job_id: CollectionJobId) -> Result<Url, Error> {
-        Ok(self.leader_endpoint.join(&format!(
-            "tasks/{}/collection_jobs/{collection_job_id}",
-            self.task_id
-        ))?)
-    }
-}
-
 /// Construct a [`reqwest::Client`] suitable for use in a DAP [`Collector`].
 pub fn default_http_client() -> Result<reqwest::Client, Error> {
     Ok(reqwest::Client::builder()
-        .redirect(reqwest::redirect::Policy::none())
```

Comment:
What's the effect of removing the redirect policy?

Reply:
It was only needed for pre-draft-04, when we expected 303 See Other responses, and needed to capture the Location header for multiple subsequent requests. Since we don't expect any redirects from DAP servers, it should have no impact, but if a sufficiently weird DAP aggregator returned a redirect response, we would now follow it instead of trying to parse an empty response or bailing on an unexpected status code.

Reply:
Yep, makes sense!